.github/workflows/ruby.yml

```diff
@@ -18,9 +18,8 @@ jobs:
     - name: Set up Ruby
       uses: ruby/setup-ruby@v1
       with:
-        ruby-version: 2.6.6
-    - name: Install dependencies
-      run: bundle install
+        ruby-version: 2.7.6 # blocked from 3.0 see https://github.com/ualbertalib/pushmi_pullyu/issues/251
+        bundler-cache: true # runs 'bundle install' and caches installed gems automatically
     - name: Lint with RuboCop
```

Comment:
I was getting an error in CI that I couldn't reproduce in development.  Turns out that the rubocop versions were different because we don't check in the Gemfile.lock.  We don't check in Gemfile.lock for ruby gems because that would make dependencies more rigid than we would want them to be.

Reply:
I ran into this error too, glad to have read you had this issue earlier 😄 

I see that you include the ``Gemfile.lock`` in this PR, will it be a problem making the dependency more rigid as you mentioned?

Reply:
I changed my mind on this after discussion on Dev Edmonton Slack.  Shane contributed to that discussion.  Even the thinking in bundler has changed about this and committing Gemfile.lock is best practice.  If you need more flexible dependencies there are other ways.  For example in the Gemfile or running test in a matrix against a variety of Gemfile.lock.

Reply:
Sorry I missed that explanation earlier 😄